mautrix_signal/user.py

```diff
@@ -309,25 +316,33 @@ async def _sync_group_v2(self, group: GroupV2, create_portals: bool) -> None:
 
     async def _sync_contacts(self) -> None:
         create_contact_portal = self.config["bridge.autocreate_contact_portal"]
-        for contact in await self.bridge.signal.list_contacts(self.username):
-            try:
-                await self.sync_contact(contact, create_contact_portal)
-            except Exception:
-                self.log.exception(f"Failed to sync contact {contact.address}")
+        try:
+            for contact in await self.bridge.signal.list_contacts(self.username):
+                try:
+                    await self.sync_contact(contact, create_contact_portal)
+                except Exception:
+                    self.log.exception(f"Failed to sync contact {contact.address}")
+        except Exception as e:
+            await handle_auth_failure(self, e)
```

Comment:
These catches should be in `sync_groups` and `sync_contacts` that are already just catch wrappers